aspnet/migration/http-modules/sample/Asp.Net5/src/Asp.Net5/Middleware/HttpContextDemoMiddleware.cs

```diff
@@ -0,0 +1,320 @@
+
+            // ASP.NET 4 - HttpContext.Request.InputStream
+            // Unlike reading the form, reading the raw body is not from a buffer.
+            // So you can only do this once per request.
+            // 
+            // If you need to read the body multiple times per request, see
+            // http://stackoverflow.com/questions/31389781/read-body-twice-in-asp-net-5/
+
+            //>16
+            string inputBody;
+            using (var reader = new System.IO.StreamReader(
+                httpContext.Request.Body, System.Text.Encoding.UTF8))
+            {
+                inputBody = reader.ReadToEnd();
+            }
+            //<16
+
+            // Use this middleware as a handler, so no call to 
+            // await _next.Invoke(aspnet5HttpContext);
+
+            // ==================================================================
+            // Response
+
+            // ASP.NET 4 - HttpContext.Response.Status  
+            // ASP.NET 4 - HttpContext.Response.StatusDescription  
+            //>30
+            httpContext.Response.StatusCode = (int)System.Net.HttpStatusCode.OK;
+            //<30
+
+            // ASP.NET 4 - HttpContext.Response.ContentEncoding and HttpContext.Response.ContentType  
+            //>31
+            // using Microsoft.Net.Http.Headers;
+            var mediaType = new MediaTypeHeaderValue("application/json");
+            mediaType.Encoding = System.Text.Encoding.UTF8;
+            httpContext.Response.ContentType = mediaType.ToString();
+            //<31
+
+            // ASP.NET 4 - HttpContext.Response.ContentType only 
+            //>32
+            httpContext.Response.ContentType = "text/html";
+            //<32
+
+            // ----------------------
+            // ASP.NET 4 - HttpContext.Response.Output  
+            //
+            // If you want to serve up the contents of a file, see
+            // http://docs.asp.net/en/latest/fundamentals/request-features.html#middleware-and-request-features
+            //
+            //>33
+            string responseContent = GetResponseContent();
+            await httpContext.Response.WriteAsync(responseContent);
+            //<33
+        }
+
+        //>41
+        // using Microsoft.AspNet.Http.Headers;
+        // using Microsoft.Net.Http.Headers;
+
+        private Task SetHeaders(object context)
+        {
+            var httpContext = (HttpContext)context;
+
+            // Set header with single value
+            httpContext.Response.Headers["ResponseHeaderName"] = "headerValue";
+
+            // Set header with multiple values
+            string[] responseHeaderValues = new string[] { "headerValue1", "headerValue1" };
+            httpContext.Response.Headers["ResponseHeaderName"] = responseHeaderValues;
+
+            // Translating ASP.NET 4's HttpContext.Response.RedirectLocation  
+            httpContext.Response.Headers["Location"] = "http://www.newurl.com";
```

Comment:
And please check any other "fake URL" uses and make sure they're example.com/org/net.